internal/metrics/metrics.go

```diff
@@ -0,0 +1,29 @@
+package metrics
+
+import (
+	"context"
+	"time"
+)
+
+// Query is the filter used to retrieve metrics data.
+type Query interface {
+	Filter(key, value string)
+	Query() string
+}
+
+// AlignReduce is the type to enumerate allowed combinations of per series
+// aligner and cross series reducer.
+type AlignReduce int32
+
+// Series aligner and cross series reducer types (for latency).
+const (
+	Align99Reduce99 AlignReduce = 1
+	Align95Reduce95             = 2
+	Align50Reduce50             = 3
+)
+
+// Client represents a wrapper around the Cloud Monitoring package.
+type Client interface {
+	Latency(ctx context.Context, query Query, startTime time.Time, alignReduceType AlignReduce) (float64, error)
```

Comment:
accept `offset time.Duration`

Reply:
I have thought about this, and I believe it makes more sense from the user of this interface to pass a startTime (`time.Time`) value since that would likely be the last time the metrics were checked. 

If the implementation of, say prometheus, uses duration, the implementation should take care of converting the startTime value to a `time.Duration`.

Reply:
> I have thought about this, and I believe it makes more sense from the user of this interface to pass a startTime (time.Time) value since that would likely be the last time the metrics were checked.

last time I recall our design was that the operator checks last N minutes; not "since last time checked". Storing such info probably requires additional storage? I also don't see a downside of looking at last N minute window.

Reply:
AFAIK, we didn't discuss exactly how to estimate the time interval between metrics checks. Both ways should work fine since I think, probabilistically, there might not be any difference in the results for slightly different times. The time intervals might even be similar in many (most) cases.

The easiest/simpler one to implement for now is to check last N minutes though. Let's do that.